JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -1,698 +1,695 @@
-				 base_breaks_y(summaryStat) +
-				 base_breaks_x(summaryStat$groupingVariable)
-								
-			image <- .beginSaveImage(options$plotWidth, options$plotHeight)
-			print(p)
-			content <- .endSaveImage(image)
-
-			descriptivesPlot[["data"]] <- content
-
-			descriptivesPlotList[[var]] <- descriptivesPlot
-			
-		}
-		
-	} else {
+.ttestIndependentSamplesNormalityTest <- function(dataset, options, perform) {
+    if (options$normalityTests == FALSE) return(NULL)
+    
+    normalityTests <- list("title" = "Test of Normality (Shapiro-Wilk)")
+    
+    ## these are the table fields associated with the normality test
+    fields <- list(
+        list(name = "dep", type = "string", title = "", combine = TRUE), 
+        list(name = "lev", type = "string", title = ""),
+        list(name = "W", title = "W", type = "number", format = "sf:4;dp:3"),
+        list(name = "p", title = "p", type = "number", format = "dp:3;p:.001")
+    )
+    
+    normalityTests[["schema"]] <- list(fields = fields)
+    
+    footnotes <- .newFootnotes()
+    .addFootnote(footnotes, symbol = "<em>Note.</em>",
+                 text = "Significant results indicate a deviation from normality")
+    
+    ## for a independent t-test, we need to check both group vectors for normality
+    normalityTests.results <- list()
+    
+    variables <- options$variables
+    factor <- options$groupingVariable
+    levels <- levels(dataset[[.v(factor)]])
 
-		for (var in .indices(options$variables)) {
+    if (length(variables) == 0) variables = "."
+    if (length(levels) == 0) levels = c(".", ".")
+    
+    for (variable in variables) {
+        count <- 0
+        
+        ## there will be maximal two levels
+        for (level in levels) {
+            count <- count + 1
+           
+            ## if everything looks fine, and we are ready to run
+            if (perform == "run" && length(variables) > 0 && !is.null(levels)) {
+
+                ## get the dependent variable at a certain factor level
+                data <- na.omit(dataset[[.v(variable)]][dataset[[.v(factor)]] == level])
+                
+                row.footnotes <- NULL
+                error <- FALSE
+                
+                if (length(data) < 3) {
+                    err <- "Too few datapoints (N < 3) to compute statistic reliably"
+                    foot.index <- .addFootnote(footnotes, err)
+                    row.footnotes <- list(W = list(foot.index), p = list(foot.index))
+                    error <- TRUE
+                  
+                } else if (length(data) > 5000) {
+                    err <- "Too many datapoints (N > 5000) to compute statistic reliably"
+                    foot.index <- .addFootnote(footnotes, err)
+                    row.footnotes <- list(W = list(foot.index), p = list(foot.index))
+                    error <- TRUE
+                }
+                
+                ## if the user did everything correctly :)
+                if (!error) {
```

Comment:
good practice to put spaces around the exclamation mark

    if ( ! error) {

it is easy to miss the exclamation mark (depending on the editor, syntax highlighting, etc. this often isn't bad, but we like to accommodate the general case).

some people go further and do the:

    if (error == FALSE) {

to be extra explicit (but not me)